tensorboard/plugins/custom_scalar/tf_custom_scalar_dashboard/tf-custom-scalar-margin-chart-card.html

```diff
@@ -401,120 +458,135 @@ <h1>[[_titleDisplayString]]</h1>
+            const scalarEvents = data.tag_to_events[tagsObject.value];
+            const lowerBounds = data.tag_to_events[tagsObject.lower];
+            const upperBounds = data.tag_to_events[tagsObject.upper];
+
+            // Make sure that data is found for each of the tags.
+            if (_.isUndefined(scalarEvents)) {
+              tagsNotFound.push(tagsObject.value);
+              tagNotFound = true;
+            }
+            if (_.isUndefined(lowerBounds)) {
+              tagsNotFound.push(tagsObject.lower);
+              tagNotFound = true;
+            }
+            if (_.isUndefined(upperBounds)) {
+              tagsNotFound.push(tagsObject.upper);
+              tagNotFound = true;
+            }
+
+            // At least one of the tags lacks data. We terminate early because
+            // the line chart requires all 3 pieces of data (value, lower bound,
+            // and upper bound).
+            if (tagNotFound) {
+              return;
+            }
+
+            // Make sure that steps for all the lists correspond with each
+            // other. Otherwise, display an error message.
+            const obtainStep = datum => datum[1];
+            const stepsMismatch = this._findStepMismatch(
+                tagsObject,
+                scalarEvents.map(obtainStep),
+                lowerBounds.map(obtainStep),
+                upperBounds.map(obtainStep));
+            if (stepsMismatch) {
+              this.set('_stepsMismatch', stepsMismatch);
+              return;
+            }
+
+            // Create data points that the line chart can parse.
+            const obtainNumber = datum => datum[2];
+            const dataPoints = scalarEvents.map((datum, i) => ({
+              wall_time: new Date(datum[0] * 1000),
+              step: obtainStep(datum),
+              scalar: obtainNumber(datum),
+              lower: obtainNumber(lowerBounds[i]),
+              upper: obtainNumber(upperBounds[i]),
+            }));
+
+            // Compute the series name, which is based on both the run and the
+            // tag of the value.
+            const seriesName =
+                tf_custom_scalar_dashboard.generateDataSeriesName(
+                    run, tagsObject.value);
+            let series = newMapping[seriesName];
+
+            if (series) {
+              // This series already exists.
+              series.setData(dataPoints);
+            } else {
+              newMapping[seriesName] = this._createNewDataSeries(
+                  run, tagsObject.value, seriesName, dataPoints);
             }
+          });
 
-            this.set('_nameToDataSeries', newMapping);
-          } else {
-            // The user's regular expression was invalid.
-            // TODO(chihuahua): Handle this.
+          if (tagsNotFound.length) {
+            // At least 1 tag could not be found. Show an error message.
+            this.set('_tagsWithNoData', tagsNotFound);
+            return;
```

Comment:
Returning here will skip `this.set('_nameToDataSeries', newMapping);` below, which is a change from what it would do before - is that desirable?  That would mean that any missing tags will cause a whole chart not to render, versus just skipping one of the potentially multiple series in the chart.

Reply:
Ah, indeed, lets avoid returning here and show the error message. And then, we'd show the chart.